docs/contributing/doc_guide.rst

```diff
@@ -949,54 +949,160 @@ Docstring guidelines
+     1D |array_like|
+     3D |array_like|
 
-     y : bool, optional, default: True
-        Description of ``y``.
+* If the shapes and sizes of the parameters are interrelated, then
+  include that information in parentheses immediately before the type
+  information. Include a trailing comma inside the parentheses when the
+  parameter is 1D. Use ``:`` for a single dimension of arbitrary size
+  and ``...`` for an arbitrary number of dimensions of arbitrary size.
 
-  * Place type information to the right of the parameter name, along
-    with ``optional`` and/or ``keyword-only`` if necessary.
+  .. code-block:: rst
 
-  * Describe any requirements for the parameter, including preconditions
-    specified using |validate_quantities| or |particle_input|.
+     (M,) |array_like|
+     (N,) |array_like|
+     (M, N) |array_like|
+     (N, :) |array_like|
+     (M, N, ...) |array_like|
 
-  * Use the substitution ``|array_like|`` to indicate that an argument
-    should be able to be converted into an |ndarray|.
+* If the parameter can only be specific values, enclose them in curly
+  brackets. The options may be listed with the default value first,
+  sorted alphanumerically, or ordered so as to maximize readability.
 
-  * If the shapes and sizes of the parameters are interrelated, then
-    include that information in parentheses immediately before the type
-    information. Include a trailing comma inside the parentheses when
-    the parameter is 1D.
+  .. code-block::
 
-    .. code-block:: rst
+     {"classical postmodernist", "retro-futuristic"}
+     {"p+", "e-"}, default: "p+"
+     {1, 2, 3, 4}, default: 3
 
-       Parameters
-       ----------
-       a : (M,) array_like
-           First input vector.
+* If a default is given, it is not necessary to state that the
+  parameter is optional. When the default is `None`, use ``optional``
+  instead of ``default: `None```.
 
-       b : (N,) array_like
-           Second input vector.
+.. tip::
 
-       out : (M, N) ndarray, optional
-           A location where the result is stored.
+   If a particular type specification is not covered above,
+   look for conventions from the `numpydoc style guide
+   <https://numpydoc.readthedocs.io/en/latest/format.html#parameters>`__,
+   the `matplotlib documentation guide
+   <https://matplotlib.org/stable/devel/documenting_mpl.html#parameter-type-descriptions>`__,
+   or the `LSST docstring guide
+   <https://developer.lsst.io/python/numpydoc.html>`__.
+
+Parameter descriptions
+^^^^^^^^^^^^^^^^^^^^^^
+
+The **parameter description** should concisely describe the meaning of
+the parameter, as well as any requirements or restrictions on allowed
+values of the parameter (including those specified by
+|validate_quantities| or |particle_input|. The parameter description
+should not repeat information already in the type specification, but may
+include type information when:
+
+- There isn't enough space in the type specification;
```

Comment:
Does this mean to keep it within the line length limit?

Reply:
Yes.  I'll clarify this.